lib/logstash/inputs/lumberjack.rb

```diff
@@ -68,12 +68,17 @@ def register
   def run(output_queue)
     start_buffer_broker(output_queue)
 
-    while true do
+    while !stop? do
       # Wrappingu the accept call into a CircuitBreaker
       if @circuit_breaker.closed?
         connection = @lumberjack.accept # Blocking call that creates a new connection
 
         invoke(connection, @codec.clone) do |_codec, line, fields|
+          if stop?
+            connection.close
```

Comment:
@lumberjack.close is already done in `def close` which close the server socket.
`connection.close` stop reading from the client socket, this allow a bit more graceful shutdown.

See
https://github.com/logstash-plugins/logstash-input-lumberjack/pull/59/files#diff-1b9590423b15f04f215635164e7376ecR101